kernel/packages/shared/apis/SignedFetch.ts

```diff
@@ -0,0 +1,65 @@
+import { ExposableAPI } from './ExposableAPI'
+import { exposeMethod, registerAPI } from 'decentraland-rpc/lib/host'
+import { AuthChain, Authenticator } from 'dcl-crypto'
+import { ParcelIdentity } from './ParcelIdentity'
+import { userAuthentified } from 'shared/session'
+import { getIdentity } from 'shared/session'
+import { flatFetch, FlatFetchInit, FlatFetchResponse } from 'atomicHelpers/flatFetch'
+import { getTLD } from '../../config'
+
+const AUTH_CHAIN_HEADER_PREFIX = 'x-identity-auth-chain-'
+const AUTH_TIMESTAMP_HEADER = 'x-identity-timestamp'
+const AUTH_METADATA_HEADER = 'x-identity-metadata'
+
+function getAuthHeaders(
+  method: string,
+  path: string,
+  metadata: Record<string, string>,
+  chainProvider: (payload: string) => AuthChain
+) {
+  const headers: Record<string, string> = {}
+  const timestamp = Date.now()
+  const metadataJSON = JSON.stringify(metadata)
+  const payloadParts = [method.toLowerCase(), path.toLowerCase(), timestamp.toString(), metadataJSON]
+  const payloadToSign = payloadParts.join(':').toLowerCase()
+
+  const chain = chainProvider(payloadToSign)
+
+  chain.forEach((link, index) => {
+    headers[`${AUTH_CHAIN_HEADER_PREFIX}${index}`] = JSON.stringify(link)
+  })
+
+  headers[AUTH_TIMESTAMP_HEADER] = timestamp.toString()
+  headers[AUTH_METADATA_HEADER] = metadataJSON
+
+  return headers
+}
+
+@registerAPI('SignedFetch')
+export class SignedFetch extends ExposableAPI {
+  parcelIdentity = this.options.getAPIInstance(ParcelIdentity)
+
+  @exposeMethod
+  async signedFetch(url: string, init?: FlatFetchInit): Promise<FlatFetchResponse> {
+    await userAuthentified()
+    const identity = getIdentity()!
+    const path = new URL(url).pathname
+    const actualInit = {
+      ...init,
+      headers: {
+        ...getAuthHeaders(
+          init?.method ?? 'get',
+          path,
+          { sceneId: this.parcelIdentity.cid, parcel: this.getSceneData().scene.base, tld: getTLD() },
+          (payload) => Authenticator.signPayload(identity, payload)
+        ),
```

Comment:
How is this different than the ephemeral key helper? Is it compatible?

Reply:
I don't understand. What is the ephemeral key helper?

Reply:
https://github.com/decentraland/ephemeralkey#usage-1

It is the library (already used by the explorer) used to sign and validate signatures

```ts
import { getHeaders } from 'ephemeralkey'

async function fetchWithEphemeralKey(request: RequestData): Promise<any> {
  const userData = JSON.parse(
    localstorage.getItem('ephemeral-data', JSON.stringify(userData))
  )
  const headers = getHeaders(userData, request)

  return fetch(request.url, {
    method: request.method,
    headers: {
      'Content-Type': 'application/json; charset=utf-8',
      ...headers
    },
    body: request.body.toString()
  })
}

const response = await fetchWithEphemeralKey({
  url: 'market.decentraland.org/api/v1/land',
  method: 'POST',
  timestamp: Date.now(),
  body: Buffer.from(JSON.stringify({ param1: 'param1', param2: 'param2' }))
})
```

Reply:
>  (already used by the explorer)
Where?

We could start using it though. I didn't know it existed until now.

I based the authentication model on the builder server: https://github.com/decentraland/builder-server/blob/master/src/middleware/authentication.ts

Reply:
Hmm it was removed in favor of `dcl-crypto#Authenticator` @nachomazzara do you know why we are no longer using ephemeralkey to generate the keys and sign?

Reply:
I don't see it being used that much, at least not in recent projects: https://github.com/search?p=3&q=org%3Adecentraland+from+ephemeralkey&type=Code

Reply:
If I remember well, `dcl-crypto` has replaced ephemeralkey because it has the possibility to create an auth chain where ephemeral is just a key/pair. It has started as a util inside the Catalyst repo, and then it was moved to its own as a library. 

CC @marcosnc or @nchamo. They may remember if it was another reason rather than the pointed above.

Reply:
I'd say to merge this as long as it works and doesn't allow anyone to use it maliciously like @nchamo said on slack. We can figure out late the details of the headers functions. I will create a reminder.

An example of malicious request would be to execute a command in a scene that overrides the main plaza via deployment if the right person is executing the scene.

Reply:
We are forcibly including `sceneId`, `parcel`, `tld` in the signature, so this cannot be used for deployments, authenticating with comms, or the builder. And it probably cannot be used anywhere else.

Other than that, future services that allow this format should validate those parameters to be sure they are not abused. But previous services don't use this format, so we should be OK.